### PR TITLE
chore: only sync identity on address change

### PR DESCRIPTION
**Pull request by tomiir**
# Description

- Only sync identity if address has changed

## Type of change

- [X] Chore (non-breaking change that addresses non-functional tasks, maintenance, or code quality improvements)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Associated Issues

For Linear issues: 
Closes APKT-1460

# Checklist

- [X] Code in this PR is covered by automated tests (Unit tests, E2E tests)
- [X] My changes generate no new warnings
- [X] I have reviewed my own code
- [X] I have filled out all required sections
- [X] I have tested my changes on the preview link
- [ ] Approver of this PR confirms that the changes are tested on the preview link

### CODE REVIEW

File: packages/appkit/src/client.ts [hunk 1]
```diff
@@ -1403,44 +1403,46 @@ export class AppKit {
...
-    )
-
-    this.setCaipAddress(
-      `${chainNamespace}:${chainId}:${address}` as `${ChainNamespace}:${string}:${string}`,
-      chainNamespace
-    )
+    // Only update state when needed
+    if (address !== AccountController.state.address) {
+      this.setCaipAddress(
+        `${chainNamespace}:${chainId}:${address}` as `${ChainNamespace}:${string}:${string}`,
```
Comment (zoruka) [inline packages/appkit/src/client.ts:1409]:
```suggestion
        `${chainNamespace}:${chainId}:${address}` as CaipAddress,
```

File: packages/appkit/src/client.ts [hunk 2]
```diff
@@ -1403,44 +1403,46 @@ export class AppKit {
...
-        : 'eoa',
-      ChainController.state.activeChain as ChainNamespace
-    )
-
-    this.setCaipAddress(
-      `${chainNamespace}:${chainId}:${address}` as `${ChainNamespace}:${string}:${string}`,
-      chainNamespace
-    )
+    // Only update state when needed
+    if (address !== AccountController.state.address) {
```
Comment (zoruka) [inline packages/appkit/src/client.ts:1407]: Is there any chance to happen a miss match due to uppercase and lowercase characters?

**Review by zoruka — APPROVED** [review]

File: packages/appkit/src/client.ts (hunk 2 above)
Comment (tomiir) [reply packages/appkit/src/client.ts:1407]: good catch!

Comment (tomiir) [reply packages/appkit/src/client.ts]: there shouldn't change cause we don't change it anywhere, but added a lowercase jic

**Pull request merged**